pdc/apps/auth/views.py

```diff
@@ -903,3 +910,231 @@ def destroy(self, request, *args, **kwargs):
         On success, HTTP status code is 204 and the response has no content.
         """
         return super(GroupResourcePermissionViewSet, self).destroy(request, *args, **kwargs)
+
+
+class APIResourcePermissionViewSet(StrictQueryParamMixin,
+                                   ChangeSetCreateModelMixin,
+                                   ChangeSetDestroyModelMixin,
+                                   mixins.RetrieveModelMixin,
+                                   mixins.ListModelMixin,
+                                   MultiLookupFieldMixin,
+                                   viewsets.GenericViewSet):
+    """
+    This end-point provides API resource permissions.
+    """
+    queryset = models.ResourcePermission.objects.all()
+    serializer_class = serializers.ResourcePermissionSerializer
+    permission_classes = (APIPermission,)
+    lookup_fields = (
+        ('resource__name', r'[^/]+'),
+        ('permission__name', r'[^/]+'),
+    )
+    filter_class = filters.ResourcePermissionFilter
+
+    def _get_users_and_groups(self, resource_permission):
+        members = dict()
+        try:
+            group_resource_permission_list = get_list_or_404(models.GroupResourcePermission,
+                                                             resource_permission=resource_permission)
+            groups_list = [str(obj.group.name) for obj in group_resource_permission_list]
+        except Http404:
+            pass
+        # get all users
+        superusers_set = {user.username for user in get_user_model().objects.filter(is_superuser=True)}
+        users_set = {user.username for user in get_user_model().objects.filter(groups__name__in=groups_list)}
+        users_list = list(superusers_set.union(users_set))
+        members['groups'] = sorted(groups_list)
+        members['users'] = sorted(users_list)
+        return members
+
+    def list(self, request, *args, **kwargs):
```

Comment:
@ycheng-aa ; I don't agree to your ideal; because this resource don't have model by itself, and in meta class, which need to add a model, this is a problem. For example, If we add {{ResourcePermission}} or {{GroupResourcePermision}} model, they are all not meet the requirement; in {{GroupResourcePermision}}, there is no {{user}} key, and {{ResourcePermission}} is too.
For the "retrieve result", I will update it.
BTW,  If my opinion is wrong, I will change it.